libraries/ESP8266WiFi/src/ESP8266WiFiGeneric.cpp

```diff
@@ -600,6 +600,69 @@ void wifi_dns_found_callback(const char *name, CONST ip_addr_t *ipaddr, void *ca
     esp_schedule(); // resume the hostByName function
 }
 
+// Async part
+
+struct host_struct {
+    String hostname;
+    IPAddress addr;
+    err_t status; // failed, ready, in progress (callback not called)
+};
+
+static struct host_struct host_var = { "", IPADDR_ANY, ERR_TIMEOUT };
+
+void wifi_dns_found_callback_async(const char* name, CONST ip_addr_t* ipaddr, void* callback_arg);
+
+int ESP8266WiFiGenericClass::hostByNameAsync(const char* aHostname, IPAddress& aResult) {
+    if (aHostname == NULL) {
+        DEBUG_WIFI_GENERIC("[hostByNameAsync] hostname is NULL!");
+        return 0;
+    }
+    static ip_addr_t addr;
+    if (!host_var.hostname.length()) {
+        host_var.hostname = aHostname;
+        err_t err = dns_gethostbyname(host_var.hostname.c_str(), &addr, &wifi_dns_found_callback_async, &host_var);

```

Comment:
Why not `host_var.status = dns_gethostbyname...`
Because `status` is never set to `ERR_INPROGRESS` but is is needed for a test below.

(btw thanks for cleaning style and space changes)

Reply:
Done